### ISSUE TITLE
✍️ Contribution period: Hamidat Mohammed

### ISSUE THREAD

**Issue by Hamidatmohd** [ersilia-os/ersilia#1010]
### Week 1 - Get to know the community

- [X] Join the communication channels
- [X] Open a GitHub issue (this one!)
- [x] Install the Ersilia Model Hub and test the simplest model
- [x] Install Docker if needed, and test another model
- [x] Write a motivation statement to work at Ersilia
- [x] Submit your first contribution to the Outreachy site

### Week 2 - Get Familiar with Machine Learning for Chemistry

- [x] Select a model from the list suggested in GitBook
- [x] Download and serve the model via the Ersilia Model Hub to ensure it works
- [x] Open a repository on your GitHub user with all the necessary files
- [x] Select and clean a dataset of 1000 molecules (example notebook 1)
- [x] Run predictions for the molecules on the selected model and evaluate the results

### Week 3 - Validate a Model in the Wild

- [x] Find a suitable dataset with sufficient experimental results
- [x] Clean and standardize the dataset
- [x] Run predictions and calculate metrics.

### Week 4 - Prepare your final application

- [ ] Submit the final application in the Outreachy website

**Comment by Hamidatmohd** [comment]
MOTIVATION STATEMENT 
My name is Hamidat Mohammed. I am an aspiring Data scientist and AI enthusiast . I am skilled in Python and Machine learning. I have always been passionate about contributing to open source and this passion led me to apply for the outreachy project. After my initial application got approved, I went through all the available projects and this one caught my attention. I am not from a science background but I have always wanted to work in the healthcare industry. This is what motivated to apply for Ersilia after reading about what they do.  This will really help my career in the sense that I will be able to learn more about machine learning and also how to apply to chemistry, science and healthcare. I am really excited about this opportunity. During the internship, I plan to dedicate my time to learning, contributing, building and growing with the Ersilia community. And after the internship I will like switch to working in the healthcare industry especially in my country Nigeria which is a developing country and contributing to how ML and AI can help to improve healthcare. Thank you so much for this opportunity.

**Comment by Hamidatmohd** [comment]
Hello @GemmaTuron and @DhanshreeA 
Please, Find the link to my repository for task 1 in week 2 for review. Thank you so much

https://github.com/Hamidatmohd/Ersilia-Model-Evaluation/blob/main/notebooks/00_model_bias.ipynb

**Comment by Hamidatmohd** [comment]
Hello @GemmaTuron  and @DhanshreeA 
Please, find the link to my repo attached for task 2 in week 2 for review. Thank you so much

https://github.com/Hamidatmohd/Ersilia-Model-Evaluation/blob/main/notebooks/01_model_reproducibility.ipynb

**Comment by Hamidatmohd** [comment]
Hello @GemmaTuron  and @DhanshreeA 
Please find my Github link attached for my repo for task 1 week 2 for review.

https://github.com/Hamidatmohd/Ersilia-Model-Evaluation/blob/main/notebooks/00_model-bias.ipynb

**Comment by Hamidatmohd** [comment]
Hello @GemmaTuron  and @DhanshreeA 
please find the link to my repo for task 2 week 2 for your review. Thank you so much

https://github.com/Hamidatmohd/Ersilia-Model-Evaluation/blob/main/notebooks/01_model-reproducibility.ipynb

**Comment by Hamidatmohd** [comment]
Hello @GemmaTuron  and @DhanshreeA 
Please find the link attached to my repo for task 2 week 2 for your review. Thank you so much

https://github.com/Hamidatmohd/Ersilia-Model-Evaluation/blob/main/notebooks/01_model_reproducibility.ipynb

**Comment by Hamidatmohd** [comment]
Hello @GemmaTuron  Thank you so much, I will do just that.

**Comment by Hamidatmohd** [comment]
WEEK 2 
Model Selection : eos6oli

Task Objective:
1. Accessing Model eos6oli bias
2. Model eos6oli Reproducibility

Task 1: Accessing the Model eos6oli bias
* The link to the Model publication can be found here [Link](https://pubs.acs.org/doi/10.1021/acs.jcim.1c00331)
* The 1,000 Molecule Datasets Used in the Bias task were obtained from the data channel on Slack.
* The Github Repo that was created for this Task can be found [here](https://github.com/Hamidatmohd/Ersilia-Model-Evaluation/tree/main/notebooks)
* The Notebook for the Data processing and Model Bias Evaluation can be found [here](https://github.com/Hamidatmohd/Ersilia-Model-Evaluation/blob/main/notebooks/00_model_bias.ipynb) and it contains further analysis and step taken to complete the task.

STEPS TAKEN FOR TASK ONE
* Firstly I installed and imported the necessary packages I needed for my analysis. 
* I performed a bit of data preprocessing by standardizing my smiles using rdkit library, checked for and dropped null values then extracting the inchi key. 
* After which I downloaded, fetched, and served the model.
* Now, I ran the dataset of 1000 molecules on the Ersilia eos6oli model and generated an output.
* I converted the output into a DataFrame and I plotted the solubility using a histogram to visually represent the distribution of the predictions.
* From the histogram, the prediction shows negative solubility values and most of these values fall within the -2 to -6 range.
* Feature extraction was done Using Morgan Fingerprints and Principal Component Analysis (PCA) was used for dimensionality reduction. The Scatter plot has been plotted and analysis done on the plot of prediction outcomes.

**Comment by Hamidatmohd** [comment]
STEPS TAKEN FOR TASK TWO

I reviewed the [publication](https://pubs.acs.org/doi/10.1021/acs.jcim.1c00331) and identified a result that I could replicate utilizing the SC2 dataset referenced in the study.

Upon locating the GitHub repository referenced in the publication, I followed the installation and usage instructions outlined in Section A of Model_Reproducibility. I proceeded by installing dependencies and the Python package associated with the model. Utilizing the SC2 dataset provided within the repository, I verified the validity of SMILES representations using the RDKit library. Subsequently, I conducted predictions with the model, preserving the output for analysis. To ensure reproducibility, I compared the model's predictions from the Soltranet model against the actual solubility values within the dataset. I then recreated a histogram observed in the publication, achieving identical results, thus confirming reproducibility. Additionally, I utilized categorized datasets available in the repository to generate sensitivity results, which aligned with those presented in the publication.
 Model_Reproducibility, utilizing the same SC2 dataset, I employed the Ersilia model to generate predictions. After saving the results, I compared them with the dataset and replicated both the histogram and sensitivity results. Remarkably, the outcomes obtained using the Ersilia model mirrored those obtained with the author's model, demonstrating consistency and reproducibility.

**Comment by GemmaTuron** [comment]
Hi @Hamidatmohd 

Thanks for the explanation, much clearer now! You did a good job, please go ahead and start working on your final application

**Comment by Hamidatmohd** [comment]
Task 3: Model Validation
The link to the Model publication can be found here [Link](https://pubs.acs.org/doi/10.1021/acs.jcim.1c00331)
The Molecule [Datasets](https://github.com/Hamidatmohd/Ersilia-Model-Evaluation/tree/main/Datasets) used in the validation task.
The Notebook for the Model Reproducibility can be found [here](https://github.com/Hamidatmohd/Ersilia-Model-Evaluation/blob/main/notebooks/02_External_Validation.ipynb). and it contains further analysis and step taken to complete the task.
STEPS TAKEN FOR TASK 3

* Dataset Sourcing and Validation
Upon conducting a brief literature review, I identified a publication containing datasets detailing molecular properties, including experimentally determined solubility. One dataset was selected as an external reference for validation purposes. Additionally, I acquired the training dataset for the soltranet model and meticulously compared its SMILES representations with those in the external dataset to ensure data integrity.

* Data Cleaning and Standardization
Among the 9955 SMILES entries in the external dataset, 368 were found to overlap with the training datasets and were consequently removed to prevent redundancy. Subsequently, I verified and standardized the remaining SMILES entries before saving the processed dataset.

* Prediction and Metric Evaluation
Predictions were generated for the training dataset using the eos6oli model, and the resultant output was saved. This output was then matched with corresponding entries in the external dataset to obtain experimental values, which were saved in a CSV file. Model validation was conducted by calculating the R2 score (coefficient of determination), yielding a score of 0.51, indicating substantial capture of variance in experimental solubility.

* Principal Component Analysis (PCA) of Morgan Fingerprint
A PCA plot was constructed using Morgan fingerprints for both the training and external validation datasets. The overlap observed between the two sets suggests the model's ability to generalize and perform effectively, even when applied to external datasets.
